tslearn/svm.py

```diff
@@ -142,57 +159,123 @@ class TimeSeriesSVC(BaseSVC):
     Marco Cuturi.
     ICML 2011.
     """
-    def __init__(self, sz, d, C=1.0, kernel="gak", degree=3, gamma="auto", coef0=0.0, shrinking=True,
-                 probability=False, tol=0.001, cache_size=200, class_weight=None, verbose=False, max_iter=-1,
+    def __init__(self, C=1.0, kernel="gak", degree=3, gamma="auto", coef0=0.0,
+                 shrinking=True, probability=True, tol=0.001, cache_size=200,
+                 class_weight=None, verbose=False, max_iter=-1,
                  decision_function_shape="ovr", random_state=None):
-        self.sz = sz
-        self.d = d
-        if kernel == "gak":
-            kernel = _kernel_func_gak(sz=sz, d=d, gamma=gamma)
-        super(TimeSeriesSVC, self).__init__(C=C, kernel=kernel, degree=degree, gamma=gamma, coef0=coef0,
-                                            shrinking=shrinking, probability=probability, tol=tol,
-                                            cache_size=cache_size, class_weight=class_weight, verbose=verbose,
-                                            max_iter=max_iter, decision_function_shape=decision_function_shape,
-                                            random_state=random_state)
+        self.C = C
+        self.kernel = kernel
+        self.degree = degree
+        self.gamma = gamma
+        self.coef0 = coef0
+        self.shrinking = shrinking
+        self.probability = probability
+        self.tol = tol
+        self.cache_size = cache_size
+        self.class_weight = class_weight
+        self.verbose = verbose
+        self.max_iter = max_iter
+        self.decision_function_shape = decision_function_shape
+        self.random_state = random_state
+
+    @property
+    def n_iter_(self):
+        warnings.warn('n_iter_ is always 1 for TimeSeriesSVC, since '
+                      'it is non-trivial to access from libsvm')
+        return 1
+
+    def _kernel_func_gak(self, x, y):
```

Comment:
This method is not used anymore, is it?

I think you use the class defined above in your code below.